### PR TITLE
implement button component

### PR DESCRIPTION
**Pull request by Aben-Bel**
here is how it looks like

image

### CODE REVIEW

File: assessment_software/src/app/button-component/button.component.css
Comment (NyraO) [inline assessment_software/src/app/button-component/button.component.css:1]: the style rules can be rewritten using tailwind css

File: assessment_software/src/app/button-component/button.component.css [hunk 1]
```diff
@@ -0,0 +1,13 @@
+.button-text {
+  font-size: 16px;
+  font-family: Inter;
+  font-weight: 500;
+  line-height: 22px;
+  word-wrap: break-word;
+}
```
Comment (NyraO) [inline assessment_software/src/app/button-component/button.component.css:7]:
```suggestion
}
   @apply text-white bg-indigo-600 py-2.5 px-5 rounded-lg border-0 hover:bg-indigo-700;

```

File: assessment_software/src/app/button-component/button.component.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+@Component({
+  selector: 'app-button',
+  templateUrl: './button.component.html',
+  styleUrls: ['./button.component.css'],
+})
+export class ButtonComponent {
+  @Input()
+  id!: string;
+  @Input() variant: 'Standard' | 'Filled' | 'Outlined' = 'Standard';
```
Comment (NyraO) [inline assessment_software/src/app/button-component/button.component.ts:11]: should there be another button variant? for a button with white background and gray text

Comment (Aben-Bel) [reply assessment_software/src/app/button-component/button.component.ts:11]: I was thinking about that and I was hoping one can change the color using the color attribute:

`<app-button variant="Standard" color="grey"> Standard</app-button>`

File: assessment_software/src/app/button-component/button.component.css
Comment (Aben-Bel) [reply assessment_software/src/app/button-component/button.component.css:1]: yes, I have done as suggested. Thank you.

File: assessment_software/src/app/button-component/button.component.css [hunk 3]
```diff
@@ -0,0 +1,15 @@
+.button-text {
+  font-size: 16px;
+  font-family: Inter;
+  font-weight: 500;
+  line-height: 22px;
+  word-wrap: break-word;
```
Comment (NyraO) [inline assessment_software/src/app/button-component/button.component.css:6]:
```suggestion
  font-size: 16px;
  font-family: Inter;
  font-weight: 500;
  line-height: 22px;
  word-wrap: break-word;
```@apply text-white bg-indigo-600 py-2.5 px-5 rounded-lg border-0 hover:bg-indigo-700;

File: assessment_software/src/app/button-component/button.component.css [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+}
+   @apply text-white bg-indigo-600 py-2.5 px-5 rounded-lg border-0 hover:bg-indigo-700;
```
Comment (NyraO) [inline assessment_software/src/app/button-component/button.component.css:8]:
```suggestion

```

File: assessment_software/src/app/button-component/button.component.css [hunk 5]
```diff
@@ -0,0 +1,15 @@
+.button-text {
+  font-size: 16px;
```
Comment (NyraO) [inline assessment_software/src/app/button-component/button.component.css:2]:
```suggestion
@apply text-white bg-indigo-600 py-2.5 px-5 rounded-lg border-0 hover:bg-indigo-700;
```

File: assessment_software/src/app/button-component/button.component.css (hunk 3 above)
Comment (NyraO) [inline assessment_software/src/app/button-component/button.component.css:6]:
```suggestion
  @apply text-white bg-indigo-600 py-2.5 px-5 rounded-lg border-0 hover:bg-indigo-700;
```

Comment (NyraO) [inline assessment_software/src/app/button-component/button.component.css:6]:
```suggestion
    @apply text-white bg-indigo-600 py-2.5 px-5 rounded-lg border-0 hover:bg-indigo-700 w-32 h-9 rounded-r;
```

File: assessment_software/src/app/button-component/button.component.html [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<button
+  class="button-primary"
```
Comment (NyraO) [inline assessment_software/src/app/button-component/button.component.html:2]:
```suggestion
  class="button-text"
```

**Review by NyraO — APPROVED** [review]

**Pull request merged**